jaxrs/src/main/java/io/airlift/jaxrs/JsonMapper.java

```diff
@@ -25,4 +33,26 @@ public JsonMapper(ObjectMapper objectMapper)
         enable(ADD_NO_SNIFF_HEADER);
         enable(INCLUDE_SOURCE_IN_LOCATION);
     }
+
+    /**
+     * Throws JsonParsingException only when Jakarta-RS container calls to deserialize given JSON value
+     * Need to distinguish between:
+     * - JsonProcessingException due to Jakarta-RS deserialization
+     * - JsonProcessingException due to some other JSON deserialization happening in resource body
+     */
+    @Override
+    public Object readFrom(Class<Object> type, Type genericType, Annotation[] annotations, MediaType mediaType, MultivaluedMap<String, String> httpHeaders, InputStream entityStream)
+            throws IOException
+    {
+        try {
+            return super.readFrom(type, genericType, annotations, mediaType, httpHeaders, entityStream);
+        }
+        catch (IOException e) {
+            // Re-throw real IO exceptions that are not due to bad JSON
+            if (!(e instanceof JsonProcessingException) && !(e instanceof EOFException)) {
```

Comment:
Copied from the previous airlift behavior here: https://github.com/airlift/airlift/commit/0e9b45932bf6753e14992bdd0fb8c40f0ab5a0d7#diff-d586e0aabebebe22e9bb3de241f2770432bae8f65dec49b748aa91741dddea7bL114